src/Exadel.CrazyPrice/Exadel.CrazyPrice.WebApi/Controllers/DiscountsController.cs

```diff
@@ -105,6 +107,13 @@ public async Task<IActionResult> GetDiscount([FromRoute] Guid id, [FromRoute] La
         public async Task<IActionResult> GetDiscounts([FromBody, CustomizeValidator(RuleSet = "SearchCriteria")] SearchCriteria searchCriteria)
         {
             searchCriteria.SearchUserId = ControllerContext.GetUserId();
+            var role = ControllerContext.GetRole().ToRoleOption();

```

Comment:
Can you add method to return role option as enum initially? for ControllerContext